File: backendworker/NHSOnline.Backend.Worker/GpSystems/Suppliers/Tpp/Models/Appointments/ViewAppointments.cs
using System;
using System.Xml.Serialization;

namespace NHSOnline.Backend.Worker.GpSystems.Suppliers.Tpp.Models.Appointments
{
    [Serializable]
    public class ViewAppointments : AbstractTppRequestModel
    {
        private ViewAppointments() { }

        public ViewAppointments(TppUserSession tppUserSession)
        {
            UnitId = tppUserSession.UnitId;
            PatientId = tppUserSession.PatientId;
            OnlineUserId = tppUserSession.OnlineUserId;
            FutureAppointments = "Y";
        }

        [XmlAttribute("patientId")]
        public string PatientId { get; set; }

        [XmlAttribute("onlineUserId")]
        public string OnlineUserId { get; set; }

        [XmlAttribute("futureAppointments")]
        public string FutureAppointments { get; set; }

        [XmlIgnore]
        public override string RequestType => "ViewAppointments";
    }
}

File: bddtests/src/main/kotlin/mocking/tpp/models/ViewAppointments.kt
package mocking.tpp.models

import javax.xml.bind.annotation.XmlAccessType
import javax.xml.bind.annotation.XmlAccessorType
import javax.xml.bind.annotation.XmlAttribute
import javax.xml.bind.annotation.XmlRootElement

@XmlRootElement(name = "ViewAppointments")
@XmlAccessorType(XmlAccessType.FIELD)
data class ViewAppointments(
    @XmlAttribute var apiVersion: String = "default apiVersion",
    @XmlAttribute var patientId: String = "default patientId",
    @XmlAttribute var onlineUserId: String = "default onlineUserId",
    @XmlAttribute var unitId: String = "default unitId",
    @XmlAttribute var uuid: String = "default uuid",
    @XmlAttribute var futureAppointments: String = "default futureAppointments"
)